WordPress/WordPressApi/WordPressOrgXMLRPCApi.swift

```diff
@@ -0,0 +1,163 @@
+
+    /**
+     Executes a XMLRPC call for the method specificied with the arguments provided.
+
+     - parameter method:  the xmlrpc method to be invoked
+     - parameter parameters: the parameters to be encoded on the request
+     - parameter success:    callback to be called on successful request
+     - parameter failure:    callback to be called on failed request
+
+     - returns:  a NSProgress object that can be used to track the progress of the request and to cancel the request. If the method
+     returns nil it's because something happened on the request serialization and the network request was not started, but the failure callback
+     will be invoked with the error specificing the serialization issues.
+     */
+    public func callMethod(method: String,
+                           parameters: [AnyObject]?,
+                           success: SuccessResponseBlock,
+                           failure: FailureReponseBlock) -> NSProgress?
+    {
+        //Encode request
+        var request: NSURLRequest? = nil
+        do {
+            request = try requestWithMethod(method, parameters: parameters)
+        } catch let encodingError as NSError {
+            failure(error: encodingError, httpResponse: nil)
+            return nil
+        }
+
+        // Create task
+        let task = session.dataTaskWithRequest(request!) { (data, urlResponse, optionalError) in
+            guard let data = data,
+                let httpResponse = urlResponse as? NSHTTPURLResponse,
+                let contentType = httpResponse.allHeaderFields["Content-Type"] as? String
+                where optionalError == nil else {
+                    if let error = optionalError {
+                        failure(error: error, httpResponse: urlResponse as? NSHTTPURLResponse)
+                    } else {
+                        failure(error: WordPressOrgXMLRPCApiError.Unknown as NSError, httpResponse: urlResponse as? NSHTTPURLResponse)
+                    }
+                    return
+            }
+
+            if ["application/xml", "text/xml"].filter({ (type) -> Bool in return contentType.hasPrefix(type)}).count == 0 {
+                failure(error: WordPressOrgXMLRPCApiError.ResponseSerializationFailed as NSError, httpResponse: httpResponse)
+                return
+            }
+
+            let decoder = WPXMLRPCDecoder(data: data)
+
+            guard !decoder.isFault(),
+                let responseXML = decoder.object() else {
+                    let decoderError = decoder.error()
+                    failure(error: decoderError,  httpResponse: httpResponse)
+                    return
+            }
+
+            success(responseObject: responseXML, httpResponse: httpResponse)
+        }
+
+        // Progress report
+        let progress = NSProgress()
+        progress.totalUnitCount = 1
+        progress.cancellationHandler = {
+            task.cancel()
+        }
+        task.resume()
+        return progress
```

Comment:
I had the same question while doing the Rest API, I wanted to make the API interface agnostic of the network stack we are using inside. That is why I opt for the progress object.